core/arch/arm/kernel/ree_fs_ta.c

```diff
@@ -247,7 +282,23 @@ static TEE_Result ree_fs_ta_read(struct user_ta_store_handle *h, void *data,
 		return TEE_ERROR_BAD_PARAMETERS;
 	if (data) {
 		dst = data; /* Hash secure buffer (shm might be modified) */
-		memcpy(dst, src, len);
+		if (handle->shdr->img_type == SHDR_ENCRYPTED_TA) {
+			/*
+			 * Encrypted TA loading only support buffered REE-FS
+			 * TA, so that encrypted TA image is read at once and
+			 * decrypted into temporary secure memory buffer.
+			 */
```

Comment:
Wouldn't it make more sense to introduce `CFG_ENCRYPT_TA` in this commit and implement a new ta_store based on the ree_fs one, similar to what was done to implement the buffered ta code? (see `#ifdef CFG_REE_FS_TA_BUFFERED ... #endif` at the end of this file).
Then rename `CFG_ENCRYPT_TA` to `CFG_REE_FS_ENCRYPTED_TA` to better match `CFG_REE_FS_TA`.

Reply:
My view of encrypted TA is just another type of user TA binary image rather than a separate ta store like REE-FS. So ideally `optee_os` should be able to load either TA image type (encrypted or plain) from REE-FS ta store. And `CFG_REE_FS_ENCRYPTED_TA` option seems to indicate that only encrypted TA can be loaded from REE-FS.

`CFG_ENCRYPT_TA` is only TA build time option to create an encrypted TA image or a plain TA image. It isn't meant for `optee_os`.

Reply:
> My view of encrypted TA is just another type of user TA binary image rather than a separate ta store like REE-FS.

Hmmm, right. Sorry I mixed things up I'm afraid ;-)

Reply:
No worries :)